flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorMasterHooksTest.java

```diff
@@ -201,7 +201,7 @@ public void testHooksAreCalledOnTrigger() throws Exception {
 		cc.addMasterHook(statefulHook2);
 
 		// trigger a checkpoint
-		assertTrue(cc.triggerCheckpoint(System.currentTimeMillis(), false));
+		assertFalse(cc.triggerCheckpoint(System.currentTimeMillis(), false).isCompletedExceptionally());
```

Comment:
Ah, sorry that's my fault. I forgot updating the java doc. I changed the behavior of `triggerCheckpoint` a bit. The return value, `CompletableFuture<CompletedCheckpoint>`, will not complete until this checkpoint is finished. It will not complete immediately after `triggerCheckpoint`. Unless there is an error occurred. It's similar to `triggerSavepoint`.
I did this because eventually the `triggerCheckpoint` will be non-blocking. It will be separated into several stages, pre-checking, initializing the id and checkpointStorageLocation, triggering snapshotting master state, and then triggering snapshotting task state. We can't judge the triggering is successful or not by a boolean return value as before.

So in this case, we can't wait for the future to complete, because it will never complete.

Reply:
If there is an error occurred immediately, the future will completed with an exception.
If the triggering is successful, we could check whether the tasks receive the `triggerCheckpoint` message or not.

Reply:
I almost missed this message.
Yes, I agree, testing case through mockito is not easy to maintain. It could be a follow-up issue (there are already some in my list). I could come back to this after the main part merged.